PCAPlot.R

```diff
@@ -0,0 +1,253 @@
+  # :param: Title - string representing the title to be shared by the 
+  #                 three side-by-side plots
+  # :param: filename - file name, including ".eps" suffix, for the plot
+  postscript(filename, width = 480, height = 480)
+  plot1 <- qplot(x = PC2, y=PC3, data = PCdata, colour = Subject) +
+    scale_colour_manual(values = colorblind_Palette) 
+  plot2 <- qplot(x = PC1, y = PC3, data = PCdata, colour = Subject) +
+    scale_colour_manual(values = colorblind_Palette) 
+  plot3 <- qplot(x = PC1, y = PC2, data = PCdata, colour = Subject) +
+    scale_colour_manual(values = colorblind_Palette) 
+  grid.arrange(plot1, plot2, plot3, ncol = 3,
+               top = title)
+  dev.off()
+}
+
+################################
+# Communication and activity PCA
+################################
+
+communication.activity <- (read.csv("communication.activity.csv"))
+communication.activity['subject'] <- 
+  (apply(communication.activity, 1, ParseSubject, 2))
+
+TimeSubj <- data.frame(communication.activity$TimeSubjectIndex)
+subj <- data.frame(communication.activity$subject)
+length <- as.numeric(dim(communication.activity)[2])
+Matrix_For_PCA <- communication.activity[,3:(length-1)]
+act.com.PCA <- prcomp(Matrix_For_PCA, scale.= TRUE)
+act.com.pca.matrix <- data.frame(act.com.PCA$x)
+
+act.com.pca.matrix["Subject"] <- subj
+write.table(act.com.PCA$rotation, "Act_Com_Loadings.csv", sep = ",")
+
+
+act.com.pca.matrix["TimeSubjectIndex"] <- TimeSubj
+
+act.com.pca.matrix["TimesOnly"] <- 
+  apply(act.com.pca.matrix, 1, ParseSubject, 1)
+
+windowed_ActCom_PCA <- 
+  subset(act.com.pca.matrix, TimesOnly %in% TimeList)
+
+###########
+# PCA Plots
+###########
+colorblind_Palette <- c("#000000", "#0072B2", "#56B4E9", "#F0E442",
+                        "#D55E00", "#CC79A7")
+
+# Activity and communication PC combinations individually
+p_1 <- qplot(x=PC2, y=PC3, data=windowed_ActCom_PCA, colour=Subject) +
+  scale_colour_manual(values=colorblind_Palette) 
+p_2 <- qplot(x=PC1, y=PC3, data=windowed_ActCom_PCA, colour=Subject) +
+  scale_colour_manual(values=colorblind_Palette)
+p_3 <- qplot(x=PC1, y=PC2, data=windowed_ActCom_PCA, colour=Subject) +
+  scale_colour_manual(values=colorblind_Palette)
+
+postscript("Facets_Act_Com_PC2PC3.eps")
+fp_1 <- p_1 + facet_grid(. ~Subject)
+grid.arrange(fp_1, ncol = 1,
+             top = paste0("Principal Components for Combined Activity",
+                          "and Communication (Visits 1 and 2)"))
+dev.off()
+
+# Activity and Energy PCA plots with PC combinations side-by-side
+PlotPCA(act.com.pca.matrix, 
+        title = paste0("Principal Components for Combined Activity and ",
+                       "Communication (All Measurements)"), 
+        filename = "act.com.PCA.PlotAll.eps")
+
+PlotPCA(windowed_ActCom_PCA, 
+        title= paste0("Principal Components for Combined Activity and ",
+                      "Communication (Visits 1 and 2)"), 
+        filename ="act.com.PCAPlot_48.eps")
+
+
+
+
```

Comment:
Should lines 89 - 101 be before the `# Activity and Energy PCA plots with PC combinations side-by-side` ? I'm a bit confused about why this is separate from the `p_2`, `p_3` in lines 63-66 in the absence of comments